### ISSUE TITLE
Silently stopped real-time filter

### ISSUE THREAD

**Issue by mielverkerken** [DocNow/twarc#319]
Hi,

I try to collect tweets over a month regarding the covid-19 outbreak with the following command.
`twarc filter 'covid19' > tweets.jsonl`
Twice the command silently stopped after +-2hours of collecting without anything in the log file.

What could be the reason and more important how could the error be logged?

**Comment by edsu** [comment]
What version of twarc are you running? We did observe a new type of http connection error which was causing requests to terminate, which was fixed in the last release v1.8.1.

**Comment by mielverkerken** [comment]
I am running v1.8.1. 
Is there a way to find these HTTP connection errors? 
Would it be a solution to add a number of allowed HTTP errors as an argument?

**Comment by edsu** [comment]
Normally you will see them logged, or on stderr (usually the console) if they were not handled. There already is --connection_errors, but it defaults to 0, which means allowing an unlimited number of them, as long as they were properly caught.

How are you running your twarc command? Is it in a tmux session or something like that? I will start up a similar filter command to see if the same thing happens to me.

**Comment by edsu** [comment]
Ok, I've started up the same command and will wait to see what happens. If you are able to attach your log here I can take a look to see if there are any hints about what might have happened.

**Comment by mielverkerken** [comment]
I just ran the command in the background on a server.
`twarc --log test.log filter 'covid19' > /dev/null &`

Log file: [test.log](https://github.com/DocNow/twarc/files/4465130/test.log)

**Comment by edsu** [comment]
Still going here after 3 hours. Can you try to not put yours into the background with &?

**Comment by edsu** [comment]
Also, can you tell me what version of Python are you using?

    python -V

Also what version of requests?

    pip3 show requests

**Comment by mielverkerken** [comment]
I am using twarc with `Python 3.6.9`

```bash
> pip3 show requests

Name: requests
Version: 2.23.0
Summary: Python HTTP for Humans.
Home-page: https://requests.readthedocs.io
Author: Kenneth Reitz
Author-email: me@kennethreitz.org
License: Apache 2.0
Location: /home/groep32/.local/lib/python3.6/site-packages
Requires: certifi, chardet, idna, urllib3
```

I started the command again with `nohup`. 
`nohup twarc --log test.log filter 'covid19' > /dev/null &`

**Comment by edsu** [comment]
... still going here after 6 hours ... (running in a tmux session)

**Comment by mielverkerken** [comment]
Thanks @edsu, nohup did the deal. So apparently the process got killed after a log out of my ssh session.

**Comment by edsu** [comment]
Let's leave this open though, because I like your idea of logging when this happens, if possible.

**Comment by edsu** [comment]
A fix for this (adding a log message when terminated) was just released as part of v1.8.2. Thanks for taking the time to file the issue ticket!
